assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/status/SW360StatusReporterParameters.java

```diff
@@ -0,0 +1,139 @@
+package org.eclipse.sw360.antenna.frontend.compliancetool.sw360.status;
+
+import org.eclipse.sw360.antenna.frontend.compliancetool.main.AntennaComplianceToolOptions;
+import org.eclipse.sw360.antenna.frontend.stub.cli.AbstractAntennaCLIOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Class responsible for parsing status reporter parameters
+ * and creating the {@code InfoRequest} the status reporter
+ * uses.
+ */
+class SW360StatusReporterParameters {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SW360StatusReporterParameters.class);
+
+    /**
+     * The long prefix to identify a parameter indication the desired info parameter of the reporter.
+     */
+    static final String REPORTER_PARAMETER_PREFIX = AbstractAntennaCLIOptions.SWITCH_PREFIX + "-info" + AbstractAntennaCLIOptions.PARAMETER_IDENTIFIER;
+
+    /**
+     * The long parameter prefix to determine the output format of the reporter
+     */
+    static final String OUTPUT_FORMAT_PREFIX_LONG = AbstractAntennaCLIOptions.SWITCH_PREFIX + "-output";
+
+    /**
+     * The short parameter prefix to determine the output format of the reporter
+     */
+    static final String OUTPUT_FORMAT_PREFIX_SHORT = AbstractAntennaCLIOptions.SWITCH_PREFIX + "o";
+
+    /**
+     * The default output format for the reporter output
+     */
+    private static final String DEFAULT_OUTPUT_FORMAT = "CSV";
+
+    /**
+     * Creates a string representing the info parameter of an{@link InfoRequest}
+     * @param parameters set of parameters to be parsed
+     * @return parsed {@code InfoRequest} with parsed additional parameters
+     * @throws IllegalArgumentException when invalid or misconfigured parameters are given or are missing.
+     */
+    static String getInfoParameterFromParameters(Set<String> parameters) {
+        if (parameters.isEmpty()) {
+            LOGGER.error("No parameters provided for the status reporter.");
+            LOGGER.info(helpMessage());
+            throw new IllegalArgumentException("No parameters provided for the status reporter.");
+        }
+        if (parameters.stream().filter(p -> p.contains(REPORTER_PARAMETER_PREFIX)).count() != 1) {
+            LOGGER.error("Too many information requests were made in this status report. ");
+            LOGGER.info(helpMessage());
+            throw new IllegalArgumentException("Too many information requests were made in this status report. ");
+        }
+        return parameters.stream().filter(p -> p.contains(REPORTER_PARAMETER_PREFIX)).findFirst().orElse("");
+    }
+
+    /**
```

Comment:
Parsing of parameters is pretty inefficient as for each parameter the whole set is traversed. Alternatively, the set could be processed once and transformed into a map with key-value pairs. From the map it is then trivial to find the value of a specific parameter. During this transformation, it is also possible to detect duplicate parameter keys.